dual_net.py

```diff
@@ -134,6 +139,45 @@ def should_log(i):
                     logger.add_summary(weight_summaries, global_step)
             self.save_weights()
 
+    def validate(self, tf_records, batch_size=128, init_from=None, num_steps=1000):
+        """Compute only the cost components for a set of tf_records, ideally a
+        holdout set, and report them to an 'eval' subdirectory of the logs.
+        """
+        cost_tensor_names = ['policy_cost', 'value_cost', 'l2_cost',
+                             'combined_cost']
+        if self.logdir is None:
+            print("Error, trainer not initialized with a logdir.")
+            return
+
+        logdir = self.logdir + '-eval'
```

Comment:
why not a subdirectory? logs/train and logs/eval sound reasonable to me.

Also, I don't feel like I have a good sense of what the difference between "validation" and "evaluation" means. Let's figure out which term is the right one to use, and stick with it.

Reply:
Validate = compute error on holdout set
Evaluation = model vs model playoff match used by AGZ.